src/autometrics/settings.py

```diff
@@ -72,5 +96,57 @@ def get_settings() -> AutometricsSettings:
     """Get the current settings."""
     global settings
     if settings is None:
-        return init_settings()
+        settings = init_settings()
     return settings
+
+
+def validate_settings(settings: AutometricsSettings):
+    """Ensure that the settings are valid. For example, we don't support Prometheus exporter with OpenTelemetry tracker."""
+    if settings["exporter"]:
+        if settings["tracker"] == TrackerType.OPENTELEMETRY and not isinstance(
+            settings["exporter"], MetricReader
+        ):
+            if settings["exporter"]["type"] == "prometheus":
+                raise ValueError(
+                    "Prometheus exporter is not supported with OpenTelemetry tracker"
+                )
+        if settings["tracker"] == TrackerType.PROMETHEUS:
+            if isinstance(settings["exporter"], MetricReader):
+                raise ValueError(
+                    "OpenTelemetry exporter is not supported with Prometheus tracker"
+                )
+            if (
+                settings["exporter"]["type"] in ["otlp-proto-http", "otlp-proto-grpc"]
+            ) or (isinstance(settings["exporter"], MetricReader)):
+                raise ValueError(
+                    "OTLP exporter is not supported with Prometheus tracker"
+                )
+
+
+def get_exporter_settings(exporter_options: Dict[str, Any]) -> ExporterOptions:
+    """Get the exporter settings from the user supplied options."""
+    if exporter_options["type"] == "prometheus-client":
+        return PrometheusExporterOptions(
+            type="prometheus-client",
+            address=exporter_options["address"],
+            port=exporter_options["port"],
+            prefix=exporter_options["prefix"],
+        )
+    if ["otlp-proto-http", "otlp-proto-grpc"].count(exporter_options["type"]) > 0:
+        return OTLPExporterOptions(
+            type=exporter_options["type"],
+            endpoint=exporter_options["endpoint"],
```

Comment:
most of these are marked as Optional so it's ok if they are just forwarded as None's. i'm pretty sure they all also have defaults/not requred

Reply:
ah i see what's happening - i had to add the whole get_exporter_settings function to make it type typeddicts correctly but this now requires all fields to be filled. i'll look into refactoring this in a more sane way

Reply:
right on, your latest changes fixed it! as far as i can tell, the option names are the same between our library and the otel one for the otlp exporter, so it might be good to drop a link to their documentation somewhere for available options? if any thing just for our future selves